src/Conversion/ONNXToTOSA/ONNXToTOSACommon.cpp

```diff
@@ -30,6 +30,274 @@ using namespace mlir;
 namespace onnx_mlir {
 namespace tosa {
 
+static int64_t multiplyDims(llvm::ArrayRef<int64_t> dims, int64_t res = 1) {
+  for (auto dim : dims) {
+    if (ShapedType::isDynamic(dim)) {
+      return ShapedType::kDynamicSize;
+    }
+    res = res * dim;
+  }
+  return res;
+}
+
+static int64_t countDynamicDims(llvm::ArrayRef<int64_t> dims) {
+  int64_t count = 0;
+  for (auto dim : dims)
+    if (ShapedType::isDynamic(dim))
+      ++count;
+  return count;
+}
+
+// Lowers Gather operators to a sequence of TOSA ops.
+// This Code is mostly the same as TF to TOSA.
+llvm::Optional<Value> convertGatherOp(PatternRewriter &rewriter, Location loc,
```

Comment:
Seeing that all of this comes from TF, should we add some copyright? (That's more of a question for @ljfitz I guess, but if you already know that the answer is "no", you can resolve this comment

Reply:
It depends - @maxbartel would the code/descriptions you took from TF be covered by `Copyright (c) 2021 Arm Limited.`? If not, and what has survived has significant copy-pasted text still from TF then we indeed need to think how this should be represented. A link to the file(s) you took material from would help me understand.

Reply:
The link is already in the description, but here it is [link](https://github.com/tensorflow/tensorflow/blob/57d2fe48f550bf23d3b87fa396a587e2d2052ed6/tensorflow/compiler/mlir/tosa/transforms/legalize_common.cc#L3719)

Reply:
I added the copyright!